### DISCUSSION TITLE
How can I exclude a directory and its subdirectories?

### DISCUSSION THREAD

**Discussion by andradei** [BurntSushi/ripgrep#2128] [Q&A] [answered]
I'm trying to search from root, but excluding some directories like `/proc` and `/sys`. But can't find the right glob patter.

Here's what I've tried in order to remove the `/proc` dir:

- `rg -g'!/proc` firefox /
- `rg -g'!proc/` firefox /
- `rg -g'!proc/*` firefox /
- `rg -g'!proc/**` firefox /

But `/proc` and its subdirectories come up in all results.

---

My `rg` is aliased as such:
```
alias rg="rg -g '!{**/node_modules/*,**/.git/*}' --hidden --follow"
```

**Comment by andradei** [comment, marked as answer]
The `--follow` was conflicting with the exclusion of `/proc` since that dir isn't really a dir or a file (it is treated differently by Linux).

I ended up running this to exclude `/proc` and other things (notice that `--no-follow` makes it work):

`rg -g '!{/proc,/sys,**/*.js,**/*.css,/home,**/*.py}' --no-follow firefox /`

**Reply by mbana** [reply]
> The `--follow` was conflicting with the exclusion of `/proc` since that dir isn't really a dir or a file (it is treated differently by Linux).
> 
> I ended up running this to exclude `/proc` and other things (notice that `--no-follow` makes it work):
> 
> `rg -g '!{/proc,/sys,**/*.js,**/*.css,/home,**/*.py}' --no-follow firefox /`

Thanks for this!

I tend to now just do:

```sh
alias rg='rg --no-follow --glob "!{/proc,/sys,$(go env GOPATH),**/.git/*}"'
alias fd='fd --exclude /proc --exclude /sys --exclude $(go env GOPATH)'
```
